proxies/openfeature-nodejs/handlers/location.ts

```diff
@@ -0,0 +1,258 @@
+        }
+    })
+    return parsedParams
+}
+
+const getOpenFeatureVariable = async (
+    openFeatureClient: OFClient,
+    params: any[]
+): Promise<DVCVariable> => {
+    const [user, key, defaultValue, type] = params
+    if (type === 'boolean') {
+        return dvcVariableFromEvaluationDetails(
+            await openFeatureClient.getBooleanDetails(key, defaultValue as boolean, user),
+            defaultValue
+        )
+    } else if (type === 'number') {
+        return dvcVariableFromEvaluationDetails(
+            await openFeatureClient.getNumberDetails(key, defaultValue as number, user),
+            defaultValue
+        )
+    } else if (type === 'JSON') {
+        return dvcVariableFromEvaluationDetails(
+            await openFeatureClient.getObjectDetails(key, defaultValue as JsonValue, user),
+            defaultValue
+        )
+    } else if (type === 'string') {
+        return dvcVariableFromEvaluationDetails(
+            await openFeatureClient.getStringDetails(key, defaultValue as string, user),
+            defaultValue
+        )
+    } else {
+        throw new Error('Invalid default value type')
+    }
+}
+
+/**
+ * Fake DVCVariable Class so that the variable type reporting works correctly
+  */
+class DVCVariable implements DVCVariableInterface {
+    constructor(
+        public key: string,
+        public value: DVCVariableInterface['value'],
+        public defaultValue: DVCVariableInterface['value'],
+        public isDefaulted: DVCVariableInterface['isDefaulted'],
+        public type: DVCVariableInterface['type']) {}
+}
+
+/**
+ * Convert OpenFeature EvaluationDetails to DVCVariable
+ */
+const dvcVariableFromEvaluationDetails = <T extends FlagValue>(
+    evalDetails: EvaluationDetails<T>,
+    defaultValue: T
+): DVCVariable => {
+    let varType: string = typeof evalDetails.value
+    if (varType === 'object') {
+        varType = 'JSON'
+    }
+
+    if (evalDetails.errorCode) {
+        console.log(`error code: ${evalDetails.errorCode}, throw error: ${evalDetails.errorMessage}`)
+        if (evalDetails.errorMessage.includes('Missing parameter:')) {
+            throw new Error(evalDetails.errorMessage)
+        }
+    }
+
+    const dvcVar = new DVCVariable(
+        evalDetails.flagKey,
+        evalDetails.value as DVCVariable['value'],
+        defaultValue as DVCVariable['value'],
+        evalDetails.reason !== StandardResolutionReasons.TARGETING_MATCH,
+        // Capitalize first letter of type
+        (varType.charAt(0).toUpperCase() + varType.slice(1)) as DVCVariable['type']
+    )
+    console.log(`dvcVar: ${JSON.stringify(dvcVar)}`)
+    return dvcVar
+}
+
+const invokeCommand = (
+    entity: DataStoreClient | DVCUser | DVCVariable | any,
+    command: string,
+    params: ParsedParams,
+) => {
+    console.log(`invoking command ${command} on ${typeof entity} with params ${JSON.stringify(params)}`)
+    const dataStoreClient = entity as DataStoreClient
+    if (command === 'variable') {
```

Comment:
This is a bit different too, basically only calls to `variable` are passed through the OpenFeature client, all others are sent to the DVCClient